File: analyze_spacing.py
import os
import re

html_files = [
    'cultureroute.html',
    'index.html',
    'about.html',
    'energysaver.html',
    'workshop.html'
]

total_matches = 0

regex1 = re.compile(r'([\u4e00-\u9fa5])\s+([a-zA-Z0-9%])')
regex2 = re.compile(r'([a-zA-Z0-9%])\s+([\u4e00-\u9fa5])')

for file in html_files:
    file_path = os.path.join(os.getcwd(), file)
    if not os.path.exists(file_path):
        continue
        
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
        
    # Split by tags to only check text nodes
    text_nodes = []
    is_inside_tag = False
    current_text = ""
    
    for char in content:
        if char == '<':
            is_inside_tag = True
            if current_text:
                text_nodes.append(current_text)
                current_text = ""
        elif char == '>':
            is_inside_tag = False
        elif not is_inside_tag:
            current_text += char
            
    if current_text:
        text_nodes.append(current_text)
        
    file_matches = 0
    for text in text_nodes:
        file_matches += len(regex1.findall(text))
        file_matches += len(regex2.findall(text))
        
    print(f"{file}: {file_matches} potential matches")
    total_matches += file_matches

print(f"Total: {total_matches} potential matches")

File: analyze_spacing.js
const fs = require('fs');
const path = require('path');

const htmlFiles = [
    'cultureroute.html',
    'index.html',
    'about.html',
    'energysaver.html',
    'workshop.html'
];

let totalMatches = 0;

htmlFiles.forEach(file => {
    const content = fs.readFileSync(path.join(__dirname, file), 'utf-8');
    // Basic regex to find text nodes roughly:
    // This is just for research, not for the final replacement.
    // Match Chinese char, space, English/Number
    const regex1 = /([\u4e00-\u9fa5])\s+([a-zA-Z0-9])/g;
    // Match English/Number, space, Chinese char
    const regex2 = /([a-zA-Z0-9])\s+([\u4e00-\u9fa5])/g;
    
    // To be safe, we shouldn't match inside tags.
    // A better way is to split by tags and only process text.
    let textNodes = [];
    let isInsideTag = false;
    let currentText = "";
    
    for (let i = 0; i < content.length; i++) {
        const char = content[i];
        if (char === '<') {
            isInsideTag = true;
            if (currentText) {
                textNodes.push(currentText);
                currentText = "";
            }
        } else if (char === '>') {
            isInsideTag = false;
        } else if (!isInsideTag) {
            currentText += char;
        }
    }
    if (currentText) textNodes.push(currentText);

    let fileMatches = 0;
    textNodes.forEach(text => {
        const m1 = text.match(regex1);
        const m2 = text.match(regex2);
        if (m1) fileMatches += m1.length;
        if (m2) fileMatches += m2.length;
    });
    
    console.log(`${file}: ${fileMatches} potential matches`);
    totalMatches += fileMatches;
});

console.log(`Total: ${totalMatches} potential matches`);
